Alerting: Add option to show inactive alerts in alert list panel (#96888)
grafana/grafana: docs/sources/panels-visualizations/visualizations/alert-list/index.md

Filter alerts by the selected folder. Only alerts from dashboards in this folder are displayed.

### Show inactive

> Filter alerts with no instances. Prometheus alerts will have 0 instances when in the Normal (inactive) state. By default these are hidden, if you choose you can show them by toggling this switch.

Reviewer: Some edits to make this clearer and style guidelines
```suggestion
Filter for Prometheus alerts with no instances. Prometheus alerts have zero (0) instances when in the Normal (inactive) state and these are hidden by default. You can choose to show them by toggling this switch.
```

> Filter for Prometheus alerts with no instances. Prometheus alerts have zero (0) instances when in the Normal (inactive) state and these are hidden by default. You can choose to show them by toggling this switch. If the **Alert instance label** filter is configured, it takes precedence and inactive alert rules are hidden.

Reviewer 2: Toggle to show alert rules with no alert instances. 

Alert rules have zero (0) alert instances when in **Normal**(inactive) state and are hidden by default.

Reviewer 2: I am not sure what this means? If you have configured the **Alert instance label** filter, it takes precedence and inactive alert rules are hidden.

what does it take precedence over? If you've configured the filter then the toggle doesn't work?

Author: This comment above might explain the logic behind it a little more too - https://github.com/grafana/grafana/pull/96888#discussion_r1854630925

Ultimately I trialled both behaviors here (one where we ignore the instance label filter and this one, where we honor it). Considering the other filters again now, there is already some level of precedence which is implied; for example if someone filters by an alert rule name, and then also applies an instance label filter, the instance label filter will win.

In this case it might be over complicating things to explain that how it reacts to the Alert instance label filter?

> Filter for Prometheus alerts with no instances. Prometheus alerts have zero (0) instances when in the Normal (inactive) state and these are hidden by default. You can choose to show them by toggling this switch. If the Alert instance label filter is configured, it will take precedence and inactive alert rules will be hidden.

Reviewer: Just a couple final edits, but otherwise this looks good
```suggestion
Filter for Prometheus alerts with no instances. Prometheus alerts have zero (0) instances when in the Normal (inactive) state and these are hidden by default. You can choose to show them by toggling this switch. If the **Alert instance label** filter is configured, it takes precedence and inactive alert rules are hidden.
```

| Alert instance label | Filter alert instances using [label](ref:alert-label) querying. For example,`{severity="critical", instance=~"cluster-us-.+"}`. |
| Datasource | Filter alerts from the selected data source. |
| Folder | Filter alerts by the selected folder. Only alerts from dashboards in this folder are displayed. |

> | Show alerts with 0 instances | Filter for alert rules with no instances. Alert rules with 0 (zero) instances are hidden by default. You can choose to show them by toggling this switch. As these rules have no instances, they will remain hidden if the **Alert instance label** filter is configured. |

Reviewer: ```suggestion
| Show alerts with 0 instances | Filter for alert rules with no instances. Alert rules with 0 (zero) instances are hidden by default. You can choose to show them by toggling this switch. Because these rules have no instances, they remain hidden if the **Alert instance label** filter is configured. |
```
